hooks/react-server/useSiteNavigation.tsx

```diff
@@ -1,3 +1,81 @@
-import useBaseSiteNavigation from '@/hooks/useBaseSiteNavigation';
+import { useTranslations } from 'next-intl';
+import type { RichTranslationValues } from 'next-intl';
+import { useMemo } from 'react';
 
-export default useBaseSiteNavigation;
+import { siteNavigation } from '@/next.json.mjs';
+import type { NavigationEntry, NavigationKeys } from '@/types';
+
+// These are mapped navigation entries. Navigation Entries can have sub-entries
+type MappedItems = {
+  text: ReturnType<ReturnType<typeof useTranslations>['rich']>;
+  link: string;
+  key: string;
+  level: number;
+  items: MappedItems[];
+};
+
+// Provides Context replacement for variables within the Link. This is also something that is not going
+// to happen in the future with `nodejs/nodejs.dev` codebase
+const replaceLinkWithContext = (link: string, context: RichTranslationValues) =>
+  Object.entries(context).reduce(
+    (finalLink, [find, replace]) =>
+      finalLink.replace(
+        `{${find}}`,
+        typeof replace === 'string' ? replace : ''
+      ),
+    link
+  );
+
+const useSiteNavigation = () => {
+  const t = useTranslations();
+
+  const mapNavigationEntries = (
+    entries: Record<string, NavigationEntry>,
+    context?: Record<string, RichTranslationValues>,
+    level = 0
+  ): MappedItems[] => {
+    const getContext = (key: string) => (context && context[key]) || {};
+
+    const getFormattedMessage = (translationId: string, key: string) =>
+      t.rich(translationId, getContext(key));
+
+    return Object.entries(entries).map(([key, item]) => ({
+      text: getFormattedMessage(item.translationId, key),
+      link: replaceLinkWithContext(item.link, getContext(key)),
+      items: item.items
+        ? mapNavigationEntries(item.items, context, level + 1)
+        : [],
+      level,
+      key: key,
```

Comment:
```suggestion
    return Object.entries(entries).map(([key, { translationId, link, items }]) => ({
      text: getFormattedMessage(translationId, key),
      link: replaceLinkWithContext(link, getContext(key)),
      items: items
        ? mapNavigationEntries(items, context, level + 1)
        : [],
      level,
      key: key,
```

I think we can make it more meaningful by destructing it. `items: item.items` may seem a little confusing for someone who will see it for the first time

Reply:
Ill see if that makes sense.